homeassistant/components/modbus/base_platform.py

```diff
@@ -160,9 +161,14 @@ async def async_update(self, now=None):
             self.async_write_ha_state()
             return
 
+        # do not allow multiple active calls to the same platform
+        if self._call_active:
+            return
+        self._call_active = True
         result = await self._hub.async_pymodbus_call(
             self._slave, self._verify_address, 1, self._verify_type
         )
+        self._call_active = False
```

Comment:
How about creating a context manager to handle this?

Reply:
that is a possibility, I just saw it as much more code for a simple task, wdyt?

Reply:
Not sure. It was just an idea.

What I had in mind was s.th. like this:
```python
from contextlib import asynccontextmanager

@asynccontextmanager
async def acquire_lock():
    if self._call_active:
        return
    self._call_active = True
    try:
        yield
    finally:
        self._call_active = False
```
```python
    async with acquire_lock():
        result = await self._hub.async_pymodbus_call(
            self._slave, self._verify_address, 1, self._verify_type
        )
```

Reply:
And I learned something today, I was not aware that I could use self. in a contextManager, it looks a bit more complex. 

Having said that, I will add it just to get the PR approved 😃

Reply:
But when I add your first block (the definition) I get errors the self. is not defined. I have tried to add the block first outside the class and then inside the class.

Then I tried the tried inside/outside with async def acquire_lock(self), thinking that would declare self. but no luck.

So unless you have an idea, I am stuck. your suggestion really have a charm by not repeating code lines.

In case you do not have an idea, do you have other review comments ?

Reply:
It was just a sketch. I'll take a look after dinner.

Reply:
thanks

Reply:
you really got me searching and trying. 
```
@asynccontextmanager
async def acquire_lock(call_active):
    if call_active:
        return call_active
    call_active = True
    try:
        yield call_active
    finally:
        call_active = False
```
A modified version, where I return call_active (since python do not have "pass by reference"), but I cannot get call_active cleared, since finally do not allow return.

the use would be 
```
    with self._call_active = acquire_lock(self._call_active):
         <code>
```

Now I am hooked on getting your suggestion to work!

Reply:
I like a challenge, now I hope you like the new version.